data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexType.java

```diff
@@ -1,7 +1,38 @@
 package com.amazon.dataprepper.plugins.sink.opensearch.index;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
 public enum IndexType {
-    TRACE_ANALYTICS_RAW,
-    TRACE_ANALYTICS_SERVICE_MAP,
-    CUSTOM;
+    TRACE_ANALYTICS_RAW("trace-analytics-raw"),
+    TRACE_ANALYTICS_SERVICE_MAP("trace-analytics-service-map"),
+    CUSTOM("custom");
+
+    private final String value;
+
+    // Reverse-lookup map for getting a IndexType from a name
```

Comment:
Please use Javadoc comments on fields. IntelliJ knows to move them with the field itself.

Reply:
Cool. Done!